microcosm_flask/tests/test_context.py

```diff
@@ -5,16 +5,32 @@
 from hamcrest import assert_that, equal_to, is_
 from microcosm.api import create_object_graph
 
+from microcosm_flask.context import make_get_request_context
+
 
 def test_request_context():
     graph = create_object_graph(name="example", testing=True)
     graph.use(
         "opaque",
-        "request_context",
     )
 
     with graph.flask.test_request_context(headers={
             "X-Request-Id": "foo",
     }):
-        with graph.opaque.initialize(graph.request_context):
+        with graph.opaque.initialize(make_get_request_context()):
             assert_that(graph.opaque["X-Request-Id"], is_(equal_to("foo")))
+
+
+def test_make_get_request_context_respects_passed_headers():
+    graph = create_object_graph(name="example", testing=True)
+    graph.use(
+        "opaque",
+    )
+
+    with graph.flask.test_request_context(headers={
+            "X-Foo-Id": "foo",
+            "X-Bar-Id": "bar",
+    }):
+        with graph.opaque.initialize(make_get_request_context(["X-Bar"])):
+            assert_that(graph.opaque["X-Bar-Id"], is_(equal_to("bar")))
```

Comment:
I'm probably confused, why is one prefix working here? Neither of these are in the whitelist or the all_header list?

Reply:
`make_get_request_context` takes a list of `header_prefixes` to pull from the `request.headers`, so on the line above I say give me all the `X-Bar` headers which means those and only those headers will come back.

Reply:
Ohh, I was missing L34. Got it.